openrlhf/trainer/ppo_trainer.py

```diff
@@ -495,16 +519,27 @@ def save_logs_and_checkpoints(self, args, global_step, step_bar, logs_dict={}, c
                     for k, v in self.experience_maker.perf_stats.items():
                         self._tensorboard.add_scalar(f"perf/experience_maker/{k}", v, global_step)
 
-        # TODO: Add evaluation mechanism for PPO
-        if global_step % args.eval_steps == 0:
-            # self.evaluate(self.eval_dataloader, global_step)
-            pass
         # save ckpt
         # TODO: save best model on dev, use loss/perplexity/others on whole dev dataset as metric
         if global_step % args.save_steps == 0:
             tag = f"global_step{global_step}"
             self._save_checkpoint(args, tag, client_states)
 
+        # Run eval after check for checkpoint  save in case checkpoint save comes at the end of the job
+        if global_step % args.eval_steps == 0 and self.eval_dataloader is not None:
+            logs_dict = self.evaluate(self.eval_dataloader, global_step)
+            if self._wandb is not None and self.strategy.is_rank_0():
+                logs = {
+                    "eval/%s" % k: v
+                    for k, v in {
+                        **logs_dict,
+                        "global_step": global_step,
+                    }.items()
+                }
+                # if self.experience_maker.perf_stats is not None:
```

Comment:
unused comments?

Reply:
yeah I can remove unless you see a good reason we need perf stats for the eval experience maker